File: PMS/productui/src/Service/Productservice.js
import axios from "axios";

const API_URL = "http://localhost:8080";

class ProductService {

    saveProduct(product) {
        return axios.post(`${API_URL}/saveprod`, product);
    }

    getAllProduct() {
        return axios.get(`${API_URL}/products`);
    }

    getProductById(id) { // Renamed for clarity
        return axios.get(`${API_URL}/products/${id}`);
    }

    deleteProduct(id) {
        return axios.delete(`${API_URL}/productdelete/${id}`); // Typically, DELETE method is used for deletions
    }

    editProduct(product) {
        return axios.post(`${API_URL}/editproduct/${product.id}`, product);
    }
}

export default new ProductService();

File: PMS/Product-Backend/src/main/java/com/demo/Service/ProductService.java
package com.demo.Service;

import java.util.List;

import com.demo.Model.Product;

public interface ProductService {
	
	public Product saveproduct (Product product);
	
	public List<Product> getallproduct();
	
	public Product getproductbyid(Long id);
	
	public String deleteproduct(Long id);
	
	public Product editProduct(Product product,Long id);
	

}
